### PR TITLE
Boto3 Upgrade

### PR DESCRIPTION
**Pull request by sean-smith**
Signed-off-by: Sean Smith <seaam@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: files/default/attachVolume.py [hunk 1]
```diff
@@ -30,24 +30,23 @@
 availableDevices = [a for a in blockDevices if a not in paths]
 
 # Connect to AWS using boto
-conn = boto.ec2.connect_to_region(region)
+ec2 = boto3.client('ec2', region_name=region)
 
 # Attach the volume
 dev = availableDevices[0].replace('xvd', 'sd')
-conn.attach_volume(volumeId,instanceId,dev)
+ec2.attach_volume(VolumeId=volumeId,InstanceId=instanceId,Device=dev)
```
Comment (enrico-usai) [inline files/default/attachVolume.py]: minor: add a space after the commas

File: files/default/ec2_dev_2_volid.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import os
 import syslog
 import time
+import boto3
```
Comment (lukeseawalker) [inline files/default/ec2_dev_2_volid.py]: is "import boto.ec2" on line 3 needed?

File: recipes/base_install.rb [hunk 3]
```diff
@@ -64,6 +64,9 @@
   version '1.15.40'
 end
 
+# Install boto3
+python_package 'boto3'
```
Comment (lukeseawalker) [inline recipes/base_install.rb]: Is this really needed? It is not already in some dependency chain?

File: files/default/ec2_dev_2_volid.py (hunk 2 above)
Comment (sean-smith) [reply files/default/ec2_dev_2_volid.py]: removed

File: files/default/attachVolume.py (hunk 1 above)
Comment (sean-smith) [reply files/default/attachVolume.py]: done

File: recipes/base_install.rb (hunk 3 above)
Comment (sean-smith) [reply recipes/base_install.rb]: yes, boto2 is installed as part of `awscli`. The base AMI does not have boto3 installed.

File: files/default/ec2_dev_2_volid.py [hunk 4]
```diff
@@ -33,20 +33,19 @@
...
-conn = boto.ec2.connect_to_region(region)
+ec2 = boto3.client('ec2', region_name=region)
 
 # Poll for blockdevicemapping
-attrib = conn.get_instance_attribute(instanceId, 'blockDeviceMapping')
-devmap = attrib.get('blockDeviceMapping')
+devices = ec2.describe_instance_attribute(InstanceId=instanceId, Attribute='blockDeviceMapping').get('BlockDeviceMappings')
 x = 0
-while not devmap.has_key(dev):
+while not any(d['DeviceName'] == dev for d in devices):
```
Comment (sean-smith) [inline files/default/ec2_dev_2_volid.py]: I realize this is confusing... the new format for BlockDeviceMappings is a list of dicts. 

In light of this I renamed `devmap` to `devices` and changed the logic to find the right item in the list.

**Pull request merged**